### PR TITLE
Update Dockerfile to install dependencies with dep

### PR DESCRIPTION
**Pull request by itzg**

### CODE REVIEW

File: Dockerfile.releases [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu
+
+ARG POLLER_VERSION=0.2.40
+
+ADD https://github.com/racker/rackspace-monitoring-poller/releases/download/$POLLER_VERSION/rackspace-monitoring-poller_linux_amd64 /usr/local/bin/rackspace-monitoring-poller
+RUN chmod +x /usr/local/bin/rackspace-monitoring-poller
+
+ENTRYPOINT ["/usr/local/bin/rackspace-monitoring-poller"]
```
Comment (itzg) [inline Dockerfile.releases]: @jjbuchan I finally determined that it is something about running from within a `bash` process. When using this form of `ENTRYPOINT` it skips using a shell and was working for me.

File: Gopkg.toml [hunk 2]
```diff
@@ -63,7 +63,7 @@
 
 [[constraint]]
   name = "github.com/sirupsen/logrus"
-  version = "0.11.4"
+  version = "1.0.6"
```
Comment (itzg) [inline Gopkg.toml]: ...I even upgraded logrus hoping that would fix the shell weirdness, but didn't make a difference.

File: Dockerfile [hunk 3]
```diff
@@ -1,36 +1,29 @@
...
-RUN openssl req \
-    -new \
-    -newkey rsa:2048 \
-    -nodes \
-    -keyout key.pem \
-    -x509 \
-    -days 365 \
-    -out cert.pem \
-    -subj "/C=US/ST=Texas/L=Rackspace/O=Dis/CN=www.example.com"
+FROM ubuntu
```
Comment (itzg) [inline Dockerfile]: Added a multi-stage build here just for fun :)

Comment (jjbuchan) [reply Dockerfile]: Cool

**Pull request merged**